chain/client/src/view_client.rs

```diff
@@ -229,50 +219,52 @@ impl ViewClientActor {
             QueryRequest::ViewCode { account_id, .. } => account_id,
         };
         let shard_id = self.runtime_adapter.account_id_to_shard_id(account_id);
+        let head = self.chain.head()?;
 
-        // If we have state for the shard that we query return query result directly.
-        // Otherwise route query to peers.
-        match self.chain.get_chunk_extra(header.hash(), shard_id) {
-            Ok(chunk_extra) => {
-                let state_root = chunk_extra.state_root;
-                self.runtime_adapter
-                    .query(
-                        shard_id,
-                        &state_root,
-                        header.height(),
-                        header.raw_timestamp(),
-                        header.prev_hash(),
-                        header.hash(),
-                        header.epoch_id(),
-                        &msg.request,
-                    )
-                    .map(Some)
-                    .map_err(|e| e.to_string())
-            }
-            Err(e) => {
-                match e.kind() {
-                    ErrorKind::DBNotFoundErr(_) => {}
-                    _ => {
-                        warn!(target: "client", "Getting chunk extra failed: {}", e.to_string());
-                    }
-                }
-                // route request
-                let mut request_manager = self.request_manager.write().expect(POISONED_LOCK_ERR);
-                if Self::need_request(msg.query_id.clone(), &mut request_manager.query_requests) {
-                    let validator = self
-                        .chain
-                        .find_validator_for_forwarding(shard_id)
-                        .map_err(|e| e.to_string())?;
-                    self.network_adapter.do_send(NetworkRequests::Query {
-                        query_id: msg.query_id,
-                        account_id: validator,
-                        block_reference: msg.block_reference,
-                        request: msg.request,
-                    });
-                }
-
-                Ok(None)
+        if self.runtime_adapter.cares_about_shard(
```

Comment:
I am sorry but please do not try to change the logic here. What you did is not an equivalent transformation.

Reply:
@bowenwang1996 can you assist here to transform it equivalent, please?

Reply:
@bowenwang1996 we removed query-routing here following your take in https://github.com/near/nearcore/pull/3204/files#diff-34fd63bb882408a6cd040297ab48d277c848f434f9f72da6696aa17e86099405R212-R230